### PR TITLE
Improve connection workflow

### LINKED ISSUE #493: (fix) Connection flow state
We should address this connection state issue: https://github.com/protofire/gnosis-conditional-exchange/issues/477#issuecomment-601771709

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #493

### CODE REVIEW

File: app/src/components/common/cookies_banner/index.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ export const CookiesBanner = () => {
 
     logger.log('Loading Google Analytics...')
 
-    GoogleAnalytics.initialize(GOOGLE_ANALYTICS_ID, { gaOptions: { cookieDomain: 'auto' }, debug: true })
+    GoogleAnalytics.initialize(GOOGLE_ANALYTICS_ID, { gaOptions: { cookieDomain: 'auto' } })
```
Comment (hpmaxi) [inline app/src/components/common/cookies_banner/index.tsx:83]: This was intended only for development

File: app/src/components/modal/modal_connect_wallet/index.tsx [hunk 2]
```diff
@@ -168,10 +179,17 @@ export const ModalConnectWallet = (props: Props) => {
...
       {!context.account && (
         <ModalWrapper isOpen={isOpen} onRequestClose={onClickCloseButton} title={`Connect a Wallet`}>
-          <ButtonsWrapper>
-            <MetamaskButton disabled={!doesMetamaskExist || !acceptedTerms} />
-            <WalletConnectButton disabled={!acceptedTerms} />
-          </ButtonsWrapper>
+          {showConnectingMessage ? (
+            <>
+              <Spinner height="25px" width="25px"></Spinner>
+              <Text>Waiting for metamask connection</Text>
```
Comment (hpmaxi) [inline app/src/components/modal/modal_connect_wallet/index.tsx:185]: @gabitoesmiapodo I tried to make it prettier but failed at it

File: app/src/components/modal/modal_connect_wallet/index.tsx [hunk 3]
```diff
@@ -138,6 +142,13 @@ export const ModalConnectWallet = (props: Props) => {
     }
   }, [context, onClickCloseButton, showWalletConnectQR])
 
+  useEffect(() => {
+    if (showConnectingMessage && context.account && context.connectorName === Wallet.MetaMask) {
+      onClickCloseButton()
+      setShowConnectingMessage(false)
+    }
+  }, [context, onClickCloseButton, showConnectingMessage])
```
Comment (hpmaxi) [inline app/src/components/modal/modal_connect_wallet/index.tsx:150]: Without this, the message is shown after a connect/disconnect/connect.

File: app/src/hooks/connectedWeb3.tsx [hunk 4]
```diff
@@ -56,6 +56,8 @@ export const ConnectedWeb3: React.FC = props => {
       }
     } else if (error) {
       logger.debug(error.message)
+      localStorage.removeItem('CONNECTOR')
+      context.setConnector('Infura')
```
Comment (hpmaxi) [inline app/src/hooks/connectedWeb3.tsx:60]: When you cancel the metamask approval an error is thrown, and the `CONNECTOR` key isn't removed from the local storage, so when you refresh the modal for approval is shown again even before the `connect` button is clicked.

**Comment by gabitoesmiapodo** [comment]
Made some modifications.

Apr-28-2020 14-30-44
Apr-28-2020 14-27-58
Apr-28-2020 14-28-57

**Pull request merged**